Remove tzmigration app

millerdev opened the pull request:
## Summary

The timezone migration has been completed as a side effect of completing the forms & cases migration.

https://github.com/dimagi/commcare-hq/blob/7335500c698cf45dc2a70f77f0990c8095a54e02/corehq/apps/couch_sql_migration/progress.py#L62-L63

**Please consider carefully**: all uses of `force_phone_timezones_should_be_processed` have been removed, and anywhere that called `phone_timezones_should_be_processed()` or `phone_timezones_have_been_processed()` have been modified as if they would always return `True`.

This removal allows form processor code to be removed more easily (work in progress on that).

:tropical_fish: Review by commit.

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage

No new tests.

### QA Plan

No QA.

### Safety story

Given that the migration has been completed, it should be safe to remove this now obsolete app.

### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

millerdev committed:
Remove all references to corehq.apps.tzmigration

millerdev committed:
rm -rf corehq/apps/tzmigration

dannyroberts commented on corehq/form_processor/models.py:
```diff
@@ -499,9 +498,7 @@ def form_data(self):
         except XMLSyntaxError:
             return {}
-        # we can assume all sql domains are new timezone domains
-        with force_phone_timezones_should_be_processed():
```
Ah okay this is the line that makes me confident that this PR has no consequences. It looks like we had never explicitly set the tzmigration to complete, but that the SQL forms implementation ignores its value anyway.

dannyroberts commented:
Looks like the build errors might be real but trivial, at least the one I looked at
```
======================================================================
FAIL: corehq.apps.dump_reload.tests.test_sql_dump_load:TestSQLDumpLoadShardedModels.test_dump_load_form
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/apps/dump_reload/tests/test_sql_dump_load.py", line 191, in test_dump_load_form
    self._dump_and_load(expected_object_counts)
  File "/mnt/commcare-hq-ro/corehq/apps/dump_reload/tests/test_sql_dump_load.py", line 114, in _dump_and_load
    self.assertDictEqual(dict(expected_model_counts), dict(actual_model_counts))
AssertionError: {'form_processor.xforminstancesql': 2, 'blo[90 chars]': 1} != {'blobs.blobmeta': 2, 'form_processor.xform[37 chars]': 1}
  {'blobs.blobmeta': 2,
-  'domain_migration_flags.domainmigrationprogress': 1,
   'form_processor.xforminstancesql': 2,
   'products.sqlproduct': 1}
```

millerdev committed:
Fix failing tests

Prior to ed2c51904384a9dcf30771c4227ffa065efa029b, Domain.save() called set_tz_migration_complete(), which created a record that was accounted for by tests in corehq.apps.dump_reload.tests.test_sql_dump_load. The accounting needed to change since it was been removed.

millerdev committed:
Lint
